Framework/DataHandling/src/LoadISISNexus2.cpp

```diff
@@ -883,14 +883,14 @@ void LoadISISNexus2::loadBlock(NXDataSetTyped<int> &data, int64_t blocksize,
   int64_t final(hist + blocksize);
   while (hist < final) {
     m_progress->report("Loading data");
-    MantidVec &Y = local_workspace->dataY(hist);
+    auto &Y = local_workspace->mutableY(hist);
     Y.assign(data_start, data_end);
     data_start += m_detBlockInfo.getNumberOfChannels();
     data_end += m_detBlockInfo.getNumberOfChannels();
-    MantidVec &E = local_workspace->dataE(hist);
+    auto &E = local_workspace->mutableE(hist);
     std::transform(Y.begin(), Y.end(), E.begin(), dblSqrt);
     // Populate the workspace. Loop starts from 1, hence i-1
-    local_workspace->setX(hist, m_tof_data);
+    local_workspace->setSharedX(hist, m_tof_data);
```

Comment:
See above, can rewrite this block of code to benefit from automatic E via `setHistogram`.